qiskit_experiments/library/characterization/qubit_spectroscopy.py

```diff
@@ -23,12 +23,18 @@
 from qiskit.qobj.utils import MeasLevel
 from qiskit.utils import apply_prefix
 
-from qiskit_experiments.framework import BaseExperiment, Options
+from qiskit_experiments.framework.experiment_data import ExperimentData
+from qiskit_experiments.framework import Options
 from qiskit_experiments.curve_analysis import ParameterRepr
+from qiskit_experiments.calibration_management.update_library import Frequency
+from qiskit_experiments.calibration_management.backend_calibrations import BackendCalibrations
 from qiskit_experiments.library.characterization.resonance_analysis import ResonanceAnalysis
+from qiskit_experiments.calibration_management.base_calibration_experiment import (
+    BaseCalibrationExperiment,
+)
 
 
-class QubitSpectroscopy(BaseExperiment):
+class QubitSpectroscopy(BaseCalibrationExperiment):
```

Comment:
It feels odd to me that a characterization experiment is a `BaseCalibrationExperiment` child. I wonder about making a `RoughFrequency` subclass of `QubitSpectroscopy` that is a `BaseCalibrationExperiment`. More generally, I wonder about making `BaseCalibrationExperiment` a mix-in class that can add on the updating feature to a `BaseExperiment`. For example, I like to use Rabi as a diagnostic experiment on a qubit, so I could also see Rabi as a characterization experiment with `RoughAmplitude` as a `BaseCalibrationExperiment` subclass.

On the other, I prefer to keep the class hierarchy as minimal as possible because long chains get confusing.

Reply:
This is one of my main concerns. I have tried to make the calibration logic optional. I.e. we can still run `QubitSpectroscopy` independently of any calibration with the current PR. However, it does feel a bit funny to see calibration logic in the characterization code. I see the point about making `BaseCalibrationExperiment` a mix-in class but I also worry about making the class structure too complex to avoid confusion.

Reply:
I think
```python
class RoughFrequency(QubitSpectroscopy, BaseCalibrationExperiment):
   ...
```
is a good suggestion though this could be confusing to users (especially finding the right experiment class).

Reply:
I explore this in bb8fc79. It's not as bad as I though but you need to think carefully about the init order as we are dealing with a diamond inheritance structure here.